route_following_plugin/include/route_following_plugin.h

```diff
@@ -166,26 +159,21 @@ namespace route_following_plugin
         ros::Timer discovery_pub_timer_;
 
         // ROS service servers
-        ros::ServiceServer plan_maneuver_srv_;        
+        ros::ServiceServer plan_maneuver_srv_;  
+
+        //Queue of maneuver plans
```

Comment:
I think there might be a risk of using a strict queue since you need to walk the queue at points before a maneuver is completed, but I haven't looked at that logic yet.

Reply:
That's a fair consideration, if we need random access or simply to non-destructively walk it then a std::queue likely wouldn't match our use case.